utils/lib/openqa_api.py

```diff
@@ -0,0 +1,839 @@
+        except KeyError:
+            # no jobs found
+            pass
+        return sorted(jobs)
+
+    @staticmethod
+    def get_latest_concluded_job_ids(test_suite, history_len, version, flavor):
+        success_jobs = OpenQA.get_latest_job_ids(test_suite, version=version,
+                            result="passed",  history_len=history_len,
+                            flavor=flavor)
+
+        failed_jobs = OpenQA.get_latest_job_ids(test_suite, version=version,
+                            result="failed",
+                            history_len=history_len, flavor=flavor)
+
+        job_ids = sorted(success_jobs + failed_jobs)
+        job_ids = job_ids[-history_len:]
+        return job_ids
+
+    @staticmethod
+    def get_n_jobs_like(reference_job, n, flavor_override=None):
+        """Obtains similar n number of concluded valid jobs
+
+        :param JobData reference_job: job whose params serve as a search base
+        :param str n: number of similar jobs to obtain
+        :param str flavor: overriding flavor
+        """
+        test_suite  = reference_job.get_job_name()
+        version     = reference_job.get_job_version()
+        if flavor_override:
+            flavor = flavor_override
+        else:
+            flavor = reference_job.get_job_flavor()
+
+        latest_job_id = OpenQA.get_latest_job_id(
+            job_type=test_suite, version=version, flavor=flavor)
+        margin = n * 2 # add margin for invalid jobs
+        max_history_len = latest_job_id - reference_job.job_id + margin
+
+        job_ids = OpenQA.get_latest_concluded_job_ids(
+                    test_suite, max_history_len, version, flavor)
+        if not job_ids:
+            return []
+
+        if reference_job.job_id in job_ids:
+            ref_job_index = job_ids.index(reference_job.job_id)
+        else:
+            # try to find job with closest id before the reference job
+            # (indicating temporal closeness)
+            ref_job_distances = map(
+                lambda job_id: reference_job.job_id - job_id, job_ids)
+            ref_job_distances = filter( # remove jobs after reference job
+                lambda distance: distance > 0, ref_job_distances)
+            closest_job_id = reference_job.job_id - min(ref_job_distances)
+            ref_job_index = job_ids.index(closest_job_id)
+
+        potential_job_ids = job_ids[:ref_job_index]
+
+        relevant_jobs = []
+        for job_id in reversed(potential_job_ids):
+            job = OpenQA.get_job(job_id)
+
+            if job.is_valid():
+                relevant_jobs = [job] + relevant_jobs
+            if len(relevant_jobs) >= n:
+                break
+        return relevant_jobs
+
+
+def config_db_session(in_memory=True, debug_db=False):
+
+    if in_memory:
+        db_engine = create_engine("sqlite:///:memory:", echo=debug_db)
+        Base.metadata.create_all(db_engine)
+    else:
+        db_file = "openqa_db.sqlite"
```

Comment:
Can you make the location configurable? Maybe env variable?

Reply:
Done. Did it both as an env var and as a parameter